git_machete/client.py

```diff
@@ -1,42 +1,44 @@
-#!/usr/bin/env python
-
-from typing import Callable, Dict, Generator, List, Optional, Tuple, TypeVar
-
-from git_machete import __version__
 import datetime
-import getopt
 import io
 import itertools
 import os
-import re
 import shutil
 import sys
-import textwrap
+from typing import List, Dict, Optional, Tuple, Generator, Callable
 
 from git_machete import utils
-from git_machete.utils import dim, get_pretty_choices, bold, colored, debug, fmt, underline, flat_map, tupled, get_second, excluding, warn
-from git_machete.options import CommandLineOptions
+from git_machete.constants import (
+    DISCOVER_DEFAULT_FRESH_BRANCH_COUNT, PICK_FIRST_ROOT, UNTRACKED,
+    AHEAD_OF_REMOTE, BEHIND_REMOTE, DIVERGED_FROM_AND_OLDER_THAN_REMOTE,
+    DIVERGED_FROM_AND_NEWER_THAN_REMOTE, NO_REMOTES, IN_SYNC_WITH_REMOTE, PICK_LAST_ROOT, EscapeCodes)
 from git_machete.exceptions import MacheteException, StopTraversal
-from git_machete.docs import short_docs, long_docs
 from git_machete.git_operations import GitContext
-from git_machete.github import GitHubPullRequest, derive_pull_request_by_head, set_base_of_pull_request, is_github_remote_url,\
-    get_parsed_github_remote_url, add_assignees_to_pull_request, add_reviewers_to_pull_request, create_pull_request,\
-    derive_current_user_login, set_milestone_of_pull_request
-from git_machete.constants import AHEAD_OF_REMOTE, BEHIND_REMOTE, BOLD, DIM, DISCOVER_DEFAULT_FRESH_BRANCH_COUNT, \
-    DIVERGED_FROM_AND_NEWER_THAN_REMOTE, DIVERGED_FROM_AND_OLDER_THAN_REMOTE, ENDC, GREEN, IN_SYNC_WITH_REMOTE, \
-    NO_REMOTES, ORANGE, PICK_FIRST_ROOT, PICK_LAST_ROOT, RED, UNTRACKED, YELLOW
-
-
-# Core utils
+from git_machete.github import (
+    GitHubPullRequest, derive_pull_request_by_head, set_base_of_pull_request,
+    get_parsed_github_remote_url, is_github_remote_url, create_pull_request,
+    derive_current_user_login, set_milestone_of_pull_request,
+    add_assignees_to_pull_request, add_reviewers_to_pull_request)
+from git_machete.options import CommandLineOptions
+from git_machete.utils import (
+    get_pretty_choices, flat_map, excluding, fmt, tupled, warn, debug, bold,
+    colored, underline, dim, get_second)
 
-T = TypeVar('T')
 
 BRANCH_DEF = Tuple[str, str]
 Hash_ShortHash_Message = Tuple[str, str, str]
-initial_current_directory: Optional[str] = utils.get_current_directory_or_none() or os.getenv('PWD')
+alias_by_command: Dict[str, str] = {
+    "diff": "d",
+    "edit": "e",
+    "go": "g",
+    "log": "l",
+    "status": "s"
+}
 
 
-# Manipulation on definition file/tree of branches
+# Allowed parameter values for show/go command
+def allowed_directions(allow_current: bool) -> str:
+    current = "c[urrent]|" if allow_current else ""
+    return current + "d[own]|f[irst]|l[ast]|n[ext]|p[rev]|r[oot]|u[p]"
```

Comment:
These probably belong to cli.py too!

Reply:
alias_by_command yes, I missed that. 

allowed_directions can't be put into cli.py, because client.py also uses that and I will get circular imports. It could be moved to third files, but there isn't one that would make sense from context point of view (even utils.py seems to be not a right place, because it is focusing on stuff unrelated to cli, git etc.). If you would like me to create a new file and put this method there let me know how you would like this file to be named.

Fact that we have this dilemma is probably a signal that files should be split more, responsibilities of particular classes and methods should be reconsidered and probably divided. For now, I would recommend leaving allowed_directions as is as it would be a massive task to redesign repo.

Reply:
Okay, let's leave as is for now, I missed that